CHEAT_SHEET.md

```diff
@@ -73,3 +73,20 @@ within_rich_text("Description") do
   expect(page).to have_text "my text"
 end
 ```
+
+## Img
+
+Selectors and helpers for finding an [ARIA compliant image](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/img_role).
+
+```ruby
+# find an image
+page.find(:img, "Label") # `aria-label`, `aria-labelledby` and for img tags `alt` attributes are supported
+
+# expect displayed value
+expect(page).to have_img "Label"
+
+# using custom filters to expect a specific `src` attribute
+expect(page).to have_img("Label") { |element| element["src"].match?(/image\.png/) }
```

Comment:
I appreciate that you've shared this example. While it's true that any matcher can accept blocks like this, I had something different in mind when I suggested a `:src` filter. Maybe something based on the [:element](https://github.com/teamcapybara/capybara/blob/master/lib/capybara/selector/definition/element.rb#L8-L10) selector:

```ruby
expression_filter(:src, [String, Regexp]) do |xpath, src|
  builder(xpath).add_attribute_conditions(src: src)
end

assert_selector :img, "Label", /image\.png/
```

Reply:
If you want, I can add that for sure. I don't mind.

However my personal opinion is to stick to the filter block syntax (I assume they are more or less functionally equivalent) to not get into a situation where more and more syntactic sugar is added for more or less arbitrary attributes (although of course `src` is somewhat related to the img role).

Let me know what you prefer and I gladly update the PR.

Reply:
It would be more consistent with core capybara api, (for example see [link](https://capybara-test-helpers.netlify.app/api/selectors.html#link)) to have a expression filter.

They are not quite functionally equivalent, if I've understood how it works correctly.  An expression filter will result in a single xpath query and make only one web driver request, where a filter block will need to make an additional webdriver request to get the `src` attribute of each found element.

Reply:
Thanks for your comment. The link you pasted is for `capybara_test_helpers` and not capybara core. It is an external library.

I know there is a slight difference, however for now I believe that this difference doesn't matter as you would need quite a lot of matching nodes before the xpath query is more efficient.

As I said I am open to implementing it the way @seanpdoyle is proposing. I'd just need a definite answer from someone who has review rights in the repo.

Reply:
For what it's worth, many of the selectors documented in the link above (https://capybara-test-helpers.netlify.app/api/selectors.html#link) are defined as part of [Capybara core](https://github.com/teamcapybara/capybara/blob/52eaecea6d154b7d664b0032cd1cbcad4788fe65/lib/capybara/selector.rb#L8-L174), and have context-specific filters.

Reply:
Ah, I did not know. Thanks for the links!